include/cuco/detail/static_map.inl

```diff
@@ -64,22 +70,27 @@ void static_map<Key, Value, Scope, Allocator>::insert(InputIt first,
 
   // TODO: memset an atomic variable is unsafe
   static_assert(sizeof(std::size_t) == sizeof(atomic_ctr_type));
-  CUCO_CUDA_TRY(cudaMemsetAsync(num_successes_, 0, sizeof(atomic_ctr_type)));
+  CUCO_CUDA_TRY(cudaMemsetAsync(num_successes_, 0, sizeof(atomic_ctr_type), stream));
   std::size_t h_num_successes;
 
   detail::insert<block_size, tile_size>
-    <<<grid_size, block_size>>>(first, first + num_keys, num_successes_, view, hash, key_equal);
+    <<<grid_size, block_size, 0, stream>>>(first, first + num_keys, num_successes_,
+                                                  view, hash, key_equal);
   CUCO_CUDA_TRY(cudaMemcpyAsync(
-    &h_num_successes, num_successes_, sizeof(atomic_ctr_type), cudaMemcpyDeviceToHost));
-  CUCO_CUDA_TRY(cudaDeviceSynchronize());
+    &h_num_successes, num_successes_, sizeof(atomic_ctr_type), cudaMemcpyDeviceToHost,
+    stream));
+  // stream'd execution assumes sync not required
+  if (stream == 0)
+    CUCO_CUDA_TRY(cudaStreamSynchronize(stream)); // ensures legacy behavior
```

Comment:
This sync is _always_ necessary, otherwise the value of `h_num_successes` is undefined.

Reply:
Alternatively we need to make the `size()` function take a stream and always launch a kernel to compute the size.

Reply:
Will make the function sync always, will update function document as well.